src/ert/config/analysis_module.py

```diff
@@ -24,265 +25,63 @@ class VariableInfo(TypedDict):
-                        f" value {value!r} of type {type(value).__name__!r}"
-                    ) from e
-            else:
-                if not isinstance(var["value"], bool):
-                    raise ValueError(
-                        f"Variable {var_name} expected type {var['type']}"
-                        f" received value `{value}` of type `{type(value)}`"
-                    )
-                # When config is first read, `value` is a string
-                # that's either "False" or "True",
-                # but since bool("False") is True we need to convert it to bool.
-                if not isinstance(value, bool):
-                    value = str(value).lower() != "false"
-
-                var["value"] = var["type"](value)
-        elif var_name in self.JUST_REMOVE_KEYWORDS:
-            pass
-        else:
-            raise ConfigValidationError(
-                f"Variable {var_name!r} not found in {self.name!r} analysis module"
-            )
-
-    @property
-    def inversion(self) -> int:
-        return self.get_variable_value("IES_INVERSION")  # type: ignore
-
-    @inversion.setter
-    def inversion(self, value: int) -> None:
-        self.set_var("IES_INVERSION", value)
-
-    def get_truncation(self) -> float:
-        return self.get_variable_value("ENKF_TRUNCATION")
-
-    def localization(self) -> bool:
-        return bool(self.get_variable_value("LOCALIZATION"))
-
-    def localization_correlation_threshold(self, ensemble_size: int) -> float:
-        return correlation_threshold(
-            ensemble_size, self.get_variable_value("LOCALIZATION_CORRELATION_THRESHOLD")
-        )
-
-    def get_steplength(self, iteration_nr: int) -> float:
-        """
-        This is an implementation of Eq. (49), which calculates a suitable
-        step length for the update step, from the book:
-        Geir Evensen, Formulating the history matching problem with
-        consistent error statistics, Computational Geosciences (2021) 25:945 –970
-
-        Function not really used moved from C to keep the class interface consistent
-        should be investigated for possible removal.
+class BaseSettings(BaseModel):
+    ies_inversion: Annotated[
+        int, Field(strict=True, ge=0, le=3, title="Inversion algorithm")
+    ] = DEFAULT_IES_INVERSION
+    enkf_truncation: Annotated[
+        float,
+        Field(strict=True, ge=0.0, le=1.0, title="Singular value truncation"),
+    ] = DEFAULT_ENKF_TRUNCATION
+
+    class Config:
+        extra = Extra.forbid
+        validate_assignment = True
+
+
+class ESSettings(BaseSettings):
+    localization: Annotated[bool, Field(title="Adaptive localization")] = False
+    localization_correlation_threshold: Annotated[
+        Optional[float],
+        Field(
+            strict=True,
+            ge=0.0,
+            le=1.0,
+            title="Adaptive localization correlation threshold",
+        ),
+    ] = None
```

Comment:
This was common to both IES and ES before, can you clarify to me why the change?

Reply:
Because it is not possible to use Localization with IES, so did not make sense to be able to configure it there.

Reply:
So only valid for `STD_ENKF` then?